packages/data-stores/src/reader/queries/use-site-subscriptions-query.ts

```diff
@@ -66,15 +68,34 @@ const useSiteSubscriptionsQuery = ( {
 		}
 	}, [ hasNextPage, isFetchingNextPage, isFetching, fetchNextPage ] );
 
-	// Flatten all the pages into a single array containing all subscriptions
-	const flattenedData = data?.pages?.map( ( page ) => page.subscriptions ).flat();
+	const resultData = useMemo( () => {
+		// Flatten all the pages into a single array containing all subscriptions
+		const flattenedData = data?.pages?.map( ( page ) => page.subscriptions ).flat();
+
+		const searchFilter = ( item: SiteSubscription ) => {
+			if ( searchTerm === '' ) {
+				return true;
+			}
+
+			const searchTermLowerCase = searchTerm.toLowerCase();
```

Comment:
The lowercase operation could be moved out of the searchFilter to avoid running it on every item.

Reply:
Good point! 👍